pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -768,6 +814,85 @@ impl RemoteTimelineClient {
         Ok(())
     }
 
+    pub(crate) async fn delete_all(self: &Arc<Self>) -> anyhow::Result<()> {
```

Comment:
I think this has only a single call site; could we have docs about the precondition (has to be stopped). And all means layer files, other files and finally the index_part.json.

Reply:
Do we consider shutdown here? We probably don't, and it will become more contained after my changes soon when all tenants are shutdown within a task for each tenant.

Reply:
It tracked with task_mgr, so I suppose task mgr will try to cancel it, but cancellation is not checked. I can add such checks, but would be good to not introduce more task_mgr usage points

Reply:
Ok... Yes, probably we need to handle the "s3 doesn't work" separatedly. Resolving for now.